skrf/circuit.py

```diff
@@ -160,6 +160,17 @@ def __init__(self,
             If True, the circuit will be automatically reduced using :func:`reduce_circuit`.
             This will change the circuit connections description, affecting inner current and voltage distributions.
             Suitable for cases where only the S-parameters of the final circuit ports are of interest. Default is False.
+        **kwargs : keyword arguments
+            passed to auto_reduce method.
+            `check_duplication` kwarg controls whether to check the connections have duplicate names. Default is True.
+
+            `split_ground` kwarg controls whether to split the global ground connection to independant.
+            Default is False.
+
+            `max_nports` kwarg controls the maximum number of ports of a Network that can be reduced in circuit. If a
+            Network in the circuit has a number of ports (nports), using the Network.connect() method to reduce the
+            circuit's dimensions becomes less efficient compared to directly calculating it with Circuit.s_external.
+            This value depends on the performance of the computer and the scale of the circuit. Default is 20.
```

Comment:
What about mentioning each of these three arguments explicitly in the signature than parsing afterwards from `kwargs`?
The user would have better IDE support in this case.

Reply:
I have experimented with using TypedDict to enhance type hinting in the IDE, and the results are promising.
![image](https://github.com/user-attachments/assets/134adf86-2469-4414-a209-350d20ee7019)
For further reference, you can check out the relevant [PEP](https://peps.python.org/pep-0692/) documentation here: